File: weather_script.py
import requests
import os

def fetch_weather(city, api_key):
    url = "https://api.openweathermap.org/data/2.5/weather"
    params = {
        "q": city,
        "appid": "c502456ddce1cb7670c768853d64e7a0",
        "units": "metric"
    }
    response = requests.get(url, params=params)
    response.raise_for_status()
    return response.json()

if __name__ == "__main__":

    # Replace with your actual API key or set as environment variable
    API_KEY = os.getenv("OPENWEATHER_API_KEY", "YOUR_API_KEY_HERE")
    city = input("Enter city name: ")
    try:
        data = fetch_weather(city, API_KEY)
        print(f"Weather in {city}: {data['weather'][0]['description']}")
        print(f"Temperature: {data['main']['temp']}°C")
    except requests.HTTPError as e:
        print("Failed to fetch weather data:", e)

File: weather_script.js
const fetch = require('node-fetch');

const API_KEY = 'c502456ddce1cb7670c768853d64e7a0'; // Replace with your API key
const CITY = 'London';
const URL = `https://api.openweathermap.org/data/2.5/weather?q=${CITY}&appid=${API_KEY}&units=metric`;

async function getWeather() {
    try {
        const response = await fetch(URL);
        if (!response.ok) {
            throw new Error(`HTTP error! status: ${response.status}`);
        }
        const data = await response.json();
        console.log(`Weather in ${CITY}: ${data.weather[0].description}, ${data.main.temp}°C`);
    } catch (error) {
        console.error('Error fetching weather data:', error.message);
    }
}

getWeather();